src/main/scala/no/gdl/bookapi/repository/TranslationRepository.scala

```diff
@@ -370,6 +370,25 @@ trait TranslationRepository {
       case (Sort.ByArrivalDateAsc) => sqls.orderBy(t.dateArrived.asc, t.bookId.asc)
       case (Sort.ByArrivalDateDesc) => sqls.orderBy(t.dateArrived.desc, t.bookId.desc)
     }
+
+    def translationsWithLanguage(languageTag: LanguageTag, limit: Int, offset: Int): List[Translation] =
```

Comment:
Can all these be in one function, using querydsl? And did you split the numberOfTranslations into a separate function in order to get a count before limiting the result set? I solved this in https://github.com/GlobalDigitalLibraryio/book-api/pull/46 if you're interested.

Reply:
Rewrote db-code.